### DISCUSSION TITLE
Tab settings compared to vscode

### DISCUSSION THREAD

**Discussion by XeonG** [eclipse-theia/theia#14622] [General]
Why can't I mousescroll on the tabbar to switch tabs? where is that setting?
Where is the setting to have multiple toolbars for when there is too many tabs on the first toolbar to spill over onto a 2nd toolbar based on minimum tab width settings.. these are basics of a decent ide and seem to be missing on theia yet they are in vscode

**Comment by msujew** [comment]
Hey @XeonG,

> Why can't I mousescroll on the tabbar to switch tabs? where is that setting?

I don't think mousescroll to switch tabs is supported in VS Code, is it? At least I couldn't find a setting for that. It just scrolls horizontally through the list of tabs.

> Where is the setting to have multiple toolbars for when there is too many tabs on the first toolbar to spill over onto a 2nd toolbar based on minimum tab width settings

We are already tracking this issue in https://github.com/eclipse-theia/theia/issues/10383. There was an attempt to support this in https://github.com/eclipse-theia/theia/pull/10499 but it led to display issues and was eventually closed.

I don't really have an answer for your questions, aside from: No one has started working on that yet. The Theia IDE and the Theia framework are free open source projects without any massive corporations sitting behind the project to pay the salary of dozens of full time engineers as is the case for VS Code. While there are financial incentives in the project, but we are far less people than you would expect from a project of this scope. So please adjust your expectations for the project accordingly. If you have any contributions to make to the project, we are always happy to review them :)

**Comment by XeonG** [comment]
The first setting is the one for mousewheel switching tabs... is a few more there that aren't in theia though..

![Code_s4A1j6oW9F](https://github.com/user-attachments/assets/09b08371-d9f4-4bfd-8bdc-fcadc6139551)


I find tabs pretty important to my workflow, thought theia was a fork of vscode so was expecting more additions on top of vscode oh well

**Reply by msujew** [reply]
> The first setting is the one for mousewheel switching tabs... is a few more there that aren't in theia though..

Ah, I see, that's good to know.

> thought theia was a fork of vscode so was expecting more additions on top of vscode oh well

Theia is not a fork of VS Code (see [VSCodium](https://vscodium.com/) if you want an open source forked binary of VS Code). Instead, it's a completely separate IDE (framework) that supports the full VS Code extension API. If we were to fork VS Code, we would be unable to supply the framework part that is used to build applications such as the [Arduino IDE](https://github.com/arduino/arduino-ide) or [Mbed studio](https://os.mbed.com/studio/). The [Theia IDE](https://github.com/eclipse-theia/theia-ide) itself is just a thin wrapper around the Theia framework (i.e. this repository). Every feature in Theia is a complete reimplementation of any feature that might exist in VS Code. We likely share <1% of code, but that's mostly for API compatibility. That's what I meant with the surprisingly large scope of Theia :)

Note that people have tried forking VS Code and add their own stuff on top of it - it usually leads to a maintenance nightmare, since rebasing on the latest VS Code is almost impossible due to the amount of refactoring that happens in the upstream repository.

**Comment by XeonG** [comment]
I see, well its a shame because theia does bring the toolbar api for extensions etc to use which is something vscode team have sat on for 5years or more and tab previews is also good... though toolbar customization thing isn't really promoted in theia well, infact its not even got a default set of icons setup to get a new user started with a good base line toolbar.

is also some annoying bugs I've come across in current version so its not really stable enough  but will check back again later
